tests/test_serialize_step.py

```diff
@@ -0,0 +1,16 @@
+def test_serialize_workload_resources(step_with_resources):
+    """Must not flatten workload resource data."""
+    config = step_with_resources
+    resources = config.steps["contains kubernetes resources"].resources
+
+    assert isinstance(resources, dict), "Resources should be defined."
+    assert "cpu" in resources, "Resources should contain data."
```

Comment:
These lines serve more to demonstrate the serialized result than to actually test the process – without the fix, the serialization crashes before reaching here.